react/src/pages/participants/ParticipantsTable.tsx

```diff
@@ -45,69 +56,77 @@ const rows = [
+                    onRowUpdate: (newData, oldData) =>
+                        new Promise((resolve, reject) => {
+                            setTimeout(() => {
+                                // const dataUpdate = [...data];
+                                // const index = oldData.tableData.id;
+                                // dataUpdate[index] = newData;
+                                // setData([...dataUpdate]);
 
-                            resolve();
-                        }, 1000);
-                    })
-            }}
-            components={{
-                Toolbar: props => (
-                    <div>
-                        <MTableToolbar {...props} />
-                        <div style={{ marginLeft: '24px' }}>
-                            <Chip label="CHEO" clickable className={classes.chip} />
-                            <Chip label="SK" clickable className={classes.chip} />
-                            <Chip label="ACH" clickable className={classes.chip} />
-                            <Chip label="BCL" clickable className={classes.chip} />
-                            <Chip label="Misc." clickable className={classes.chip} />
+                                resolve();
+                            }, 1000);
+                        }),
+                }}
+                components={{
+                    Toolbar: props => (
+                        <div>
+                            <MTableToolbar {...props} />
+                            <div style={{ marginLeft: '24px' }}>
+                                <Chip label="CHEO" clickable className={classes.chip} />
+                                <Chip label="SK" clickable className={classes.chip} />
+                                <Chip label="ACH" clickable className={classes.chip} />
+                                <Chip label="BCL" clickable className={classes.chip} />
+                                <Chip label="Misc." clickable className={classes.chip} />
+                            </div>
                         </div>
-                    </div>
-                ),
-            }}
-        />
+                    ),
+                }}
+                actions={[
+                    {
+                        tooltip: 'Delete the selected participants',
+                        icon: DeleteIcon,
+                        onClick: (evt, data) => {
+                            const sampleString = (data as Participant[]).map((participant) => {return participant.participantID}).join(', ')
+                            alert(`Widthdraw all datasets and records associated with the samples: ${sampleString}`)
+                        }
+                    },
+                    {
+                        tooltip: 'Run an analysis with the selected datasets',
+                        icon: PlayArrowIcon,
+                        onClick: (evt, data) => {
+                            setActiveParticipants(data as Participant[])
```

Comment:
Apparently Material Table's second parameter passed to onClick can be either a single object or an array so I'm not sure if there's a way around this and how we know it will always be the intended type? I've seen this cast in several places but will have to investigate further.

Reply:
I have no idea either ... dealing with this was a pain. I've noticed that the return type always seems to be an array even when 1 element in selected. No idea why a single object is there but yes this will have to be solved later.